src/OpenLoco/Map/TileManager.cpp

```diff
@@ -321,9 +280,85 @@ namespace OpenLoco::Map::TileManager
 
         if (quad > 0)
         {
-            height.landHeight += quad / 2;
+            return quad / 2;
+        }
+        return 0;
+    }
+
+    /**
+     * Return the absolute height of an element, given its (x, y) coordinates
+     * remember to & with 0xFFFF if you don't want water affecting results
+     *
+     * @param x @<ax>
+     * @param y @<cx>
+     * @return height @<edx>
+     *
+     * 0x00467297 rct2: 0x00662783 (numbers different)
+     */
+    TileHeight getHeight(const Pos2& pos)
+    {
+        TileHeight height{ 16, 0 };
+        // Off the map
+        if ((unsigned)pos.x >= (Map::map_width - 1) || (unsigned)pos.y >= (Map::map_height - 1))
+            return height;
+
+        auto tile = TileManager::get(pos);
+        // Get the surface element for the tile
+        auto surfaceEl = tile.surface();
+
+        if (surfaceEl == nullptr)
+        {
+            return height;
         }
 
+        height.waterHeight = surfaceEl->water() * 16;
+        height.landHeight = surfaceEl->baseZ() * 4;
+
+        const auto slope = surfaceEl->slopeCorners();
+
+        // Subtile coords
+        const auto xl = pos.x & 0x1f;
+        const auto yl = pos.y & 0x1f;
+
+        // Slope logic:
+        // Each of the four bits in slope represents that corner being raised
+        // slope == 15 (all four bits) is not used and slope == 0 is flat
+        // If the extra_height bit is set, then the slope goes up two z-levels (this happens with one corner down with oppisite corner up)
+
+        // We arbitrarily take the SW corner to be closest to the viewer
+
+        switch (slope)
+        {
+            case SurfaceSlope::flat:
+                // Flat surface requires no further calculations.
+                break;
+
+            case SurfaceSlope::n_corner_up:
+            case SurfaceSlope::e_corner_up:
+            case SurfaceSlope::s_corner_up:
+            case SurfaceSlope::w_corner_up:
+                height.landHeight += getOneCornerUpLandHeight(xl, yl, slope);
+                break;
+
+            case SurfaceSlope::ne_side_up:
+            case SurfaceSlope::se_side_up:
+            case SurfaceSlope::nw_side_up:
+            case SurfaceSlope::sw_side_up:
+                height.landHeight += getOneSideUpLandHeight(xl, yl, slope);
+                break;
+
+            case SurfaceSlope::w_corner_dn:
+            case SurfaceSlope::s_corner_dn:
+            case SurfaceSlope::e_corner_dn:
+            case SurfaceSlope::n_corner_dn:
```

Comment:
nitpick, but would be nice to have these in the same order at the `*up` enum values